istioctl/pkg/precheck/precheck.go

```diff
@@ -310,11 +310,22 @@ func checkDestinationRuleTLS(cli kube.CLIClient, messages *diag.Messages) error
 		return err
 	}
 	checkVerify := func(tls *networking.ClientTLSSettings) bool {
-		return tls != nil && tls.CaCertificates == "" && tls.CredentialName == "" &&
-			tls.Mode != networking.ClientTLSSettings_ISTIO_MUTUAL && !tls.InsecureSkipVerify.GetValue()
+		if tls == nil {
+			return false
+		}
+		if tls.Mode == networking.ClientTLSSettings_DISABLE || tls.Mode == networking.ClientTLSSettings_ISTIO_MUTUAL {
+			return false
+		}
+		return tls.CaCertificates == "" && tls.CredentialName == "" && !tls.InsecureSkipVerify.GetValue()
 	}
 	checkSNI := func(tls *networking.ClientTLSSettings) bool {
-		return tls != nil && tls.Sni == "" && tls.Mode != networking.ClientTLSSettings_ISTIO_MUTUAL
+		if tls == nil {
+			return false
+		}
+		if tls.Mode == networking.ClientTLSSettings_DISABLE || tls.Mode == networking.ClientTLSSettings_ISTIO_MUTUAL {
+			return false
+		}
+		return tls.Sni == ""
```

Comment:
```suggestion
		return tls.Sni == "" && len(tls.SubjectAltNames) == 0 &&  !tls.GetInsecureSkipVerify().GetValue()
```

Reply:
If SNI is empty: we automatically set SNI
If SNI is empty AND SAN is empty: we automatically set SNI and SAN

So the logic I have is correct I think. The behavior is changed even if SAN is explicitly set.

Reply:
Right, san check is more strict, so just check sni for warning is ok